modules/NEON/variables.tf

```diff
@@ -0,0 +1,306 @@
+  type        = list(string)
+  default     = null
+}
+
+variable "billing_budget_notification_email_addresses" {
+  description = "A list of email addresses which will be recieving billing budget notification alerts. A maximum of 4 channels are allowed as the first element of `trusted_users` is automatically added as one of the channel. {{UIMeta group=0 order=13 updatesafe }}"
+  type        = set(string)
+  default     = []
+  validation {
+    condition     = length(var.billing_budget_notification_email_addresses) <= 4
+    error_message = "Maximum of 4 email addresses are allowed for the budget monitoring channel."
+  }
+}
+
+variable "billing_budget_pubsub_topic" {
+  description = "If true, creates a Cloud Pub/Sub topic where budget related messages will be published. Default is false. {{UIMeta group=0 order=14 updatesafe }}"
+  type        = bool
+  default     = false
+}
+
+variable "boot_disk_size_gb" {
+  description = "The size of the boot disk in GB attached to this instance.  {{UIMeta group=3 order=8 options=50,100,500 }}"
+  type        = number
+  default     = 100
+}
+
+variable "boot_disk_type" {
+  description = "Disk types for notebook instances. {{UIMeta group=3 order=7 }}"
+  type        = string
+  default     = "PD_SSD"
+}
+
+variable "create_budget" {
+  description = "If the budget should be created. {{UIMeta group=0 order=5 updatesafe }}"
+  type        = bool
+  default     = false
+}
+
+variable "create_container_image" {
+  description = "If the notebook needs to have image type as Container set this variable to true, set it to false when using dafault image type i.e. VM. {{UIMeta group=3 order=3 }}"
+  type        = bool
+  default     = false
+}
+
+variable "create_network" {
+  description = "If the module has to be deployed in an existing network, set this variable to false. {{UIMeta group=2 order=1 }}"
+  type        = bool
+  default     = true
+}
+
+variable "create_project" {
+  description = "Set to true if the module has to create a project.  If you want to deploy in an existing project, set this variable to false. {{UIMeta group=1 order=1 }}"
+  type        = bool
+  default     = true
+}
+
+variable "create_usermanaged_notebook" {
+  description = "Set to true if you want to create user managed workbench notebooks. If you want to create google managed workbench notebook, set this variable to false. {{UIMeta group=3 order=1 }}"
+  type        = bool
+  default     = true
+}
+
+variable "container_image_repository" {
+  description = "Container Image Repo, only set if creating container image notebook instance by setting `create_container_image` variable to true. {{UIMeta group=3 order=4 }}"
+  type        = string
+  default     = ""
+}
+
+variable "container_image_tag" {
+  description = "Container Image Tag, only set if creating container image notebook instance by setting `create_container_image` variable to true. {{UIMeta group=3 order=5 }}"
```

Comment:
Not safe to update?

Reply:
No because if this variable is updated then the image of the notebook is updated and the notebooks is re-created and delete the existing data.